### DISCUSSION TITLE
Can't save schema datatype

### DISCUSSION THREAD

**Discussion by alessmar** [Budibase/budibase#6972] [General]
Hi everyone,
I created a REST API datasource and I need to transform some data from text, do some calculation and output a number, I tried to change the schema from Text to Number but it's automatically set to Text whenever I send the request.

Did I miss to select something in the interface?

**Comment by melohagan** [comment]
Hi @alessmar 

Whenever you click **Send** it will typically default types to Text. 
After you edit your transformer to get the data in the format you like, you can then click **Send** and then change the types to 'Number' and **Save** your query. 
That schema should stick within your app.
